modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -0,0 +1,1229 @@
+  std::unique_ptr<ReplicatedMesh> out_meshes = dynamic_pointer_cast<ReplicatedMesh>(out_meshes_ptr);
+  std::vector<std::pair<Real, Real>> positions_inner;
+  std::vector<std::pair<Real, Real>> d_positions_outer;
+
+  std::vector<std::vector<unsigned int>> inner_index;
+  std::vector<std::vector<unsigned int>> outer_index;
+  std::vector<std::pair<Real, Real>> sub_positions_inner;
+  std::vector<std::pair<Real, Real>> sub_d_positions_outer;
+  unsigned int num_total_nodes_p;
+
+  if (mesh_type == CORNER_MESH)
+  {
+    // corner mesh has three sides that needs peripheral meshes.
+    // each element has three sub-elements, representing beginning, middle, and ending azimuthal
+    // points
+    inner_index = {{0, 1, 2}, {2, 3, 4}, {4, 5, 6}};
+    outer_index = {{0, 1, 2}, {2, 3, 4}, {4, 5, 6}};
+  }
+  else
+  {
+    // side mesh has two sides that needs peripheral meshes.
+    inner_index = {{0, 1, 2}, {2, 7, 8}};
+    outer_index = {{0, 1, 2}, {2, 7, 8}};
+  }
+
+  // extra_dist includes background and ducts.
+  // Loop to calculate the positions of the boundaries.
+  for (unsigned int i = 0; i < extra_dist.size(); i++)
+  {
+    position_setup(&positions_inner,
+                   &d_positions_outer,
+                   i == 0 ? 0.0 : extra_dist[i - 1],
+                   extra_dist[i],
+                   pitch,
+                   i);
+
+    num_total_nodes_p = (num_sectors_per_side_array[pattern] + 1) *
+                        (peripheral_duct_intervals[i] + 1); // use uniformed azi first
+    std::vector<Node *> nodes_p(num_total_nodes_p);
+    // Loop for all applicable sides that need peripherial mesh (3 for corner and 2 for edge)
+    for (unsigned int peripheral_index = 0; peripheral_index < inner_index.size();
+         peripheral_index++)
+    {
+      // Loop for beginning, middle and ending positions of a side
+      for (unsigned int vector_index = 0; vector_index < 3; vector_index++)
+      {
+        sub_positions_inner.push_back(positions_inner[inner_index[peripheral_index][vector_index]]);
+        sub_d_positions_outer.push_back(
+            d_positions_outer[outer_index[peripheral_index][vector_index]]);
+      }
+      auto mesh_p = buildReplicatedMesh(2); // create an empty 2D mesh
+      auto meshp0 = build_simple_peripheral(dynamic_pointer_cast<ReplicatedMesh>(mesh_p),
+                                            num_sectors_per_side_array[pattern],
+                                            peripheral_duct_intervals[i],
+                                            sub_positions_inner,
+                                            sub_d_positions_outer,
+                                            &nodes_p,
+                                            i);
+      ReplicatedMesh other_mesh(*meshp0);
+      // rotate the peripheral mesh to the desired side of the hexagon.
+      MeshTools::Modification::rotate(other_mesh, rotation_angle, 0, 0);
+      out_meshes->prepare_for_use();
+      other_mesh.prepare_for_use();
```

Comment:
what are you calling this for exactly?
This can trigger renumbering, partitioning etc. 
Not in this case in particular since this is replicated meshes, but it s worth leaving a comment for when this is extended to distributed meshes

Reply:
I copied this from some template. Seems not needed.